simple-backend/nlpviewer_backend/lib/utils.py

```diff
@@ -55,3 +55,23 @@ def fetch_project_check_perm(id, user, perm):
 
     return project
 
+def fetch_job_check_perm(user, perm):
+    """Fetches a job by id and check the permission.
+    
+    Fetches a job by id and check whether the user has certain permission.
+
+    Args:
+        user:
+            A User instance.
+        perm:
+            Permission to check. Example: "nlpviewer_backend.read_project"
+            
+    Returns:
+        A json response of the or forbidden or not found.
+        
+    """
+    job = get_list_or_404(Job, assignee=user)
+    
+    check_perm_project(job, user, perm)
```

Comment:
I think the permission is not so correct. The `check_perm_project` checks if the user have the permission to the project, but here you provide a job?

Reply:
Fixed

Reply:
I am not sure if you can use `check_perm_project` function, see here:

https://github.com/asyml/stave/blob/master/simple-backend/nlpviewer_backend/lib/utils.py#L6

The first parameter should be a project not a job